cloud/scope/vpc_cluster.go

```diff
@@ -1646,3 +1711,340 @@ func (s *VPCClusterScope) createSecurityGroupRuleRemote(remote infrav1beta2.VPCS
+	monitor := &vpcv1.LoadBalancerPoolHealthMonitorPrototype{
+		Delay:      ptr.To(pool.HealthMonitor.Delay),
+		MaxRetries: ptr.To(pool.HealthMonitor.Retries),
+		Timeout:    ptr.To(pool.HealthMonitor.Timeout),
+		Type:       ptr.To(string(pool.HealthMonitor.Type)),
+	}
+	if pool.HealthMonitor.Port != nil {
+		monitor.Port = pool.HealthMonitor.Port
+	}
+	if pool.HealthMonitor.URLPath != nil {
+		monitor.URLPath = pool.HealthMonitor.URLPath
+	}
+	backendPool := vpcv1.LoadBalancerPoolPrototype{
+		Algorithm:     ptr.To(string(pool.Algorithm)),
+		HealthMonitor: monitor,
+		Protocol:      ptr.To(string(pool.Protocol)),
+	}
+	// Only apply a name if one was provided (otherwise rely on generated name from VPC service).
+	if pool.Name != nil {
+		backendPool.Name = pool.Name
+	}
+
+	return backendPool
+}
+
+// getDefaultBalancerBackendPools returns a list of default Load Balancer Backend Pools for a Load Balancer.
+func (s *VPCClusterScope) getDefaultLoadBalancerBackendPools() []vpcv1.LoadBalancerPoolPrototype {
+	defaultPools := make([]vpcv1.LoadBalancerPoolPrototype, 0)
+
+	// For now, only one default pool is expected.
+	defaultPool := infrav1beta2.VPCLoadBalancerBackendPoolSpec{
+		Algorithm: infrav1beta2.VPCLoadBalancerBackendPoolAlgorithmRoundRobin,
+		HealthMonitor: infrav1beta2.VPCLoadBalancerHealthMonitorSpec{
+			Delay:   5,
+			Retries: 2,
+			Timeout: 2,
+			Type:    infrav1beta2.VPCLoadBalancerBackendPoolHealthMonitorTypeTCP,
+		},
+		// Use default backend pool service name.
+		Name:     s.GetServiceName(infrav1beta2.ResourceTypeLoadBalancerPool),
+		Protocol: infrav1beta2.VPCLoadBalancerBackendPoolProtocolTCP,
+	}
+
+	defaultPools = append(defaultPools, s.buildLoadBalancerBackendPool(defaultPool))
+	return defaultPools
+}
+
+// buildLoadBalancerListener will create a Load Balancer Listener based on the provided spec.
+func (s *VPCClusterScope) buildLoadBalancerListener(additionalListener infrav1beta2.AdditionalListenerSpec) vpcv1.LoadBalancerListenerPrototypeLoadBalancerContext {
+	listener := vpcv1.LoadBalancerListenerPrototypeLoadBalancerContext{
+		Port: ptr.To(additionalListener.Port),
+		// Default protocol to TCP.
+		Protocol: ptr.To(string(infrav1beta2.VPCLoadBalancerListenerProtocolTCP)),
+	}
+	// Override protocol if it was defined.
+	if additionalListener.Protocol != nil {
+		listener.Protocol = ptr.To(string(*additionalListener.Protocol))
+	}
+	// Set the Pool name for the listener if it was defined.
+	if additionalListener.DefaultPoolName != nil {
+		listener.DefaultPool = &vpcv1.LoadBalancerPoolIdentityByName{
+			Name: additionalListener.DefaultPoolName,
+		}
+	}
+
+	return listener
+}
+
+// getDefaultLoadBalancerListeners returns a list of default Load Balancer Listeners for a Load Balancer.
+func (s *VPCClusterScope) getDefaultLoadBalancerListeners(defaultBackendPool bool) []vpcv1.LoadBalancerListenerPrototypeLoadBalancerContext {
```

Comment:
Same comment as `getDefaultLoadBalancerBackendPools()`'s comment.

Reply:
Same as the pools, regarding additional default listeners.